### DISCUSSION TITLE
Clarifications and Questions on Data Privacy and Telemetry

### DISCUSSION THREAD

**Discussion by db-dhananjayb** [artilleryio/artillery#2225] [Artillery]
We are in the final stages of our evaluation of Artillery, and we've been impressed with the product. We have some questions related to data privacy and security that we'd appreciate clarification on.

From our understanding, during the execution of a performance test using Artillery, a variety of data points, including request and response headers, bodies, cookies, usernames, passwords, test data, and performance test results, are generated. These data are stored in our AWS account, and it seems that Artillery, as well as its associated vendors, do not have permissions to access or use this information. In essence, this means that the treatment of test artifacts in our AWS account is analogous to that of tests conducted using other tools like JMeter or k6.

Additionally, as mentioned in your documentation (https://www.artillery.io/docs/resources/telemetry), it's understood that the telemetry data does not track specific details like the number of concurrent users, response sizes, or any of the sensitive information mentioned above or mentioned in the page.

We would like to seek further clarification on two points:
1. If we do not opt out of the telemetry feature, could you provide information on how long the captured telemetry data is retained?
2. Is there a mechanism in place for users to request the deletion or anonymization of telemetry data if desired, in line with data privacy principles?

We believe that understanding these aspects is vital as we consider Artillery for our performance testing needs. Any insights you can provide will be greatly appreciated.
Also, given that we operate in the fintech domain, if there are any specific considerations or insights related to data security, privacy, or any other security/permissions related aspect that you believe are relevant in the context of our evaluation, we would greatly appreciate your guidance and expertise. Please feel free to share any additional information or considerations that you think are important for us to be aware of.

Thank you for your assistance.

**Comment by hassy** [comment]
Glad to hear you're considering Artillery @db-dhananjayb! 

> From our understanding, during the execution of a performance test using Artillery, a variety of data points, including request and response headers, bodies, cookies, usernames, passwords, test data, and performance test results, are generated. These data are stored in our AWS account, and it seems that Artillery, as well as its associated vendors, do not have permissions to access or use this information.

This is correct. Everything that could be considered sensitive such as requests or responses, usernames, passwords, IP addresses or hostnames etc, stays in your own AWS account when running tests with the Artillery CLI. Neither anyone at Artillery nor any of the vendors we depend on have a way of accessing that information.

> If we do not opt out of the telemetry feature, could you provide information on how long the captured telemetry data is retained?

Our retention period for telemetry data is currently set to 3 years but this may change in the future.

> Is there a mechanism in place for users to request the deletion or anonymization of telemetry data if desired, in line with data privacy principles?

All telemetry data is anonymized **before** being sent to our analytics endpoint. We do not have a way to tie any of the telemetry data we receive back to an individual user or a company. So there's no way to delete it because it's already anonymized before it reaches us. :)

I appreciate you asking - telemetry is very useful for us to inform product decisions, but if it's a concern, then we'd rather you use Artillery with [an opt-out of telemetry](https://www.artillery.io/docs/resources/telemetry#opting-out) than not use it at all. 

---

(The above information is accurate for the Artillery CLI. [Artillery Cloud](https://www.artillery.io/cloud) is different due to the nature of the solution i.e. it's hosted by us rather than running on-prem in a user's environment.)
